src/charm_state.py

```diff
@@ -1015,6 +1070,8 @@ def _store_state(cls, state: "CharmState") -> None:
         # Convert pydantic object to python object serializable by json module.
         state_dict["proxy_config"] = json.loads(state_dict["proxy_config"].json())
         state_dict["charm_config"] = json.loads(state_dict["charm_config"].json())
+        if state.reactive_config:
+            state_dict["reactive_config"] = json.loads(state_dict["reactive_config"].json())
```

Comment:
I'm wondering if this is an immutable config - if so, could you add this configuration to the [KeyError check](https://github.com/canonical/github-runner-operator/blob/597c1ba8a444ac77270fb29d4eedac71d6d11cfb/src/charm_state.py#L1083) in `def _check_immutable_config_change` please? (so that the upgrade charm does not throw an error)

Reply:
Can I address this in an upcoming PR? At the moment the config is not immutable, but it will probably become so once we make spawning runners reactive.

Reply:
Perfectly fine with me :D thanks!